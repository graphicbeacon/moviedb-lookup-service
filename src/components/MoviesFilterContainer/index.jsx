import React, { useState } from 'react';
import { GLOBAL_CONFIG } from 'config/constants';
import MoviesFilterForm from 'components/MoviesFilterForm/index';
import MoviesFilterResultsCard from 'components/MoviesFilterResultsCard/index';
import MoviesService from 'services/MoviesService';

const { endpoint, apiKey } = GLOBAL_CONFIG;
const moviesService = new MoviesService(apiKey, endpoint);

const MoviesFilterContainer = () => {
  const [movies, setMovies] = useState([]);

  // TODO Cover with unit tests
  const fetchMovies = search => {
    const encodedSearch = encodeURIComponent(search);
    moviesService.getMovies(encodedSearch).then(movies => {
      console.log(movies);
      setMovies(movies);
    });
  };

  return (
    <div className="container">
      <div className="row">
        <div className="col-sm">
          <MoviesFilterForm
            onUpdate={fetchMovies}
            placeholder="Enter term and search"
          />
        </div>
      </div>
      <div className="row">
        {/* TODO Cover with unit test */}
        {movies.length == 0 && (
          <div className="col-sm-12">
            <div className="alert alert-secondary">No movies to display</div>
          </div>
        )}
        {/* TODO Cover with unit test */}
        {movies.length > 0 &&
          movies.map(movieDetails => (
            <div className="col-sm-4 mb-3">
              <MoviesFilterResultsCard {...movieDetails} />
            </div>
          ))}
      </div>
    </div>
  );
};

export default MoviesFilterContainer;
