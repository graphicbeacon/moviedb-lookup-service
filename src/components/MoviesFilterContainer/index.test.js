import React from 'react';
import TestRenderer from 'react-test-renderer';
import MoviesFilterContainer from './index';

describe('<MoviesFilterContainer />', () => {
  test('should match generated snapshot', () => {
    const tree = TestRenderer.create(<MoviesFilterContainer />).toJSON();
    expect(tree).toMatchSnapshot();
  });
});
