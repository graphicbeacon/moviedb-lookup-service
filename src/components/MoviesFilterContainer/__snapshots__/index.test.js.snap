// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<MoviesFilterContainer /> should match generated snapshot 1`] = `
<div
  className="container"
>
  <div
    className="row"
  >
    <div
      className="col-sm"
    >
      <form
        className="mt-5 mb-5"
        data-testid="searchForm"
        onSubmit={[Function]}
      >
        <div
          className="input-group"
        >
          <input
            className="form-control form-control-lg"
            data-testid="searchInput"
            onChange={[Function]}
            placeholder="Enter term and search"
            type="search"
            value=""
          />
          <div
            className="input-group-append"
          >
            <button
              className="btn btn-primary"
              data-testid="searchSubmit"
              type="submit"
            >
              Search
            </button>
          </div>
        </div>
      </form>
    </div>
  </div>
</div>
`;
